js/client-mainmenu.js

```diff
@@ -1487,5 +1559,20 @@
 			return buf;
 		}
 	});
-
+	window.addEventListener("load", function () {
+		if (window.location.search.includes("challenge")) {
+			window.Storage.prefs("page", "challenge");
+			setTimeout(function () {
+			//changing url without refreshing page, ther was issue of losing data when refreshing page
+				window.history.replaceState(null, "", window.location.protocol+"//" + window.Config.routes.client+"/");
+				//log in user before sending challenge
+				new window.UserPopup({name:''}).login();
+			}, 1000);
+			let data = window.location.search.split("-");
+			let userId = data[1];
+			let format = data[2];
+			window.Storage.prefs("challenger", userId);
```

Comment:
Why is Storage.prefs necessary? This data shouldn't be needed across several sessions.